metricbeat/helper/prometheus/module.go

```diff
@@ -0,0 +1,144 @@
+package prometheus
+
+import (
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	defaultScheme = "http"
+	defaultPath   = "/metrics"
+)
+
+var (
+	// HostParser validates Prometheus URLs
+	HostParser = parse.URLHostParserBuilder{
+		DefaultScheme: defaultScheme,
+		DefaultPath:   defaultPath,
+	}.Build()
+
+	expectedFlag = flag.Bool("update_expected", false, "Update prometheus expected files")
+)
+
+// MetricSetBuilder returns a builder function for a new Prometheus metricset using the given mapping
+func MetricSetBuilder(mapping *MetricsMapping) func(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	return func(base mb.BaseMetricSet) (mb.MetricSet, error) {
+		prometheus, err := NewPrometheusClient(base)
+		if err != nil {
+			return nil, err
+		}
+		return &prometheusMetricSet{
+			BaseMetricSet: base,
+			prometheus:    prometheus,
+			mapping:       mapping,
+		}, nil
+	}
+}
+
+type prometheusMetricSet struct {
+	mb.BaseMetricSet
+	prometheus Prometheus
+	mapping    *MetricsMapping
+}
+
+func (m *prometheusMetricSet) Fetch() ([]common.MapStr, error) {
```

Comment:
Can happen in a later PR, but would be great to also use here the ReporterV2 interface.